crates/common/download/src/download.rs

```diff
@@ -450,111 +545,84 @@ mod tests {
         }
     }
 
-    // Parameters:
-    //
-    // status code
-    //
-    // bearer token boolean
-    //
-    // maybe url
-    //
-    // expected std error
-    //
-    // description
-    #[test_case(
-        200,
-        false,
-        Some("not_a_url"),
-        "builder error: relative URL without a base"
-        ; "builder error"
-    )]
-    #[test_case(
-        200,
-        true,
-        Some("not_a_url"),
-        "builder error: relative URL without a base"
-        ; "builder error with auth"
-    )]
-    #[test_case(
-        200,
-        false,
-        Some("http://not_a_url"),
-        "dns error: failed to lookup address information"
-        ; "dns error"
-    )]
-    #[test_case(
-        200,
-        true,
-        Some("http://not_a_url"),
-        "dns error: failed to lookup address information"
-        ; "dns error with auth"
-    )]
-    #[test_case(
-        404,
-        false,
-        None,
-        "404 Not Found"
-        ; "client error"
-    )]
-    #[test_case(
-        404,
-        true,
-        None,
-        "404 Not Found"
-        ; "client error with auth"
-    )]
```

Comment:
Some of these test cases are still relevant, right? Even with the new retry logic, we still fail-fast on these 4XX errors, right?

Reply:
That's right, that's my bad, I wanted to rewrite that test to remove stringification of the errors (because `Display`/`to_string()` does not stringify the error `.source()`, which can have more details about the error). Will readd and improve that test. Thanks for this catch!